lib/feature/wallet/wallet_deploy/widgets/wallet_deploy_multisig_body.dart

```diff
@@ -0,0 +1,326 @@
+              ? TextInputAction.done
+              : TextInputAction.next,
+          onSubmitted: (value) {
+            if (index != custodianControllers.length - 1) {
+              custodianFocuses[index + 1].requestFocus();
+            } else {
+              _next(context);
+            }
+          },
+          maxLength: publicKeyLength,
+          suffixIcon: _custodianSuffixIcon(controller, index),
+          inputFormatters: [
+            FilteringTextInputFormatter.deny(RegExp(r'\s')),
+            FilteringTextInputFormatter.allow(RegExp('[a-fA-F0-9]')),
+          ],
+          validator: _validatePublicKey,
+          outerActions: [
+            if (index >= minConfirmationsCount)
+              CommonIconButton.svg(
+                svg: Assets.images.trash.path,
+                buttonType: EverButtonType.secondary,
+                color: colors.alert,
+                onPressed: () => _removeCustodian(index),
+                size: CommonIconButtonSize.small,
+              ),
+          ],
+        );
+      },
+    );
+  }
+
+  Widget _custodianSuffixIcon(
+    TextEditingController controller,
+    int index,
+  ) {
+    return ValueListenableBuilder<TextEditingValue>(
+      valueListenable: controller,
+      builder: (context, value, _) {
+        final colors = context.themeStyle.colors;
+
+        return value.text.isNotEmpty
+            ? CommonIconButton.icon(
+                buttonType: EverButtonType.ghost,
+                icon: Icons.clear_rounded,
+                size: CommonIconButtonSize.medium,
+                onPressed: controller.clear,
+                color: colors.textSecondary,
+              )
+            : CommonIconButton.svg(
+                buttonType: EverButtonType.ghost,
+                svg: Assets.images.paste.path,
+                size: CommonIconButtonSize.medium,
+                onPressed: () => _pasteCustodian(index),
+                color: colors.blue,
+              );
+      },
+    );
+  }
+
+  void _next(BuildContext context) {
+    FocusScope.of(context).unfocus();
+
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      if (_formKey.currentState?.validate() ?? false) {
+        context.read<WalletDeployBloc>().add(
+              WalletDeployEvent.deployMultisig(
+                _collectValidKeys(),
+                _collectRequireConfirmations(),
+              ),
+            );
+      }
+    });
+  }
+
+  /// Get list of valid public keys
+  List<PublicKey> _collectValidKeys() => custodianControllers
+      .where((v) => _validatePublicKey(v.text) == null)
+      .map((v) => PublicKey(publicKey: v.text))
+      .toList();
+
+  int _collectRequireConfirmations() {
+    if (_validateRequireConfirmations(requireConfirmationController.text) ==
+        null) {
+      return int.parse(requireConfirmationController.text);
+    }
+
+    return defaultRequireConfirmations;
+  }
+
+  String? _validatePublicKey(String? value) {
```

Comment:
I'm not sure, because if we add this in constructor or somewhere else, we will not be able to write tests with stupid cases like `PublicKey(publicKey: '123')`.

Reply:
And if we add this validation to `PublicKey` class, it won't has access to `LocaleKeys`